From 78e5af0bb715612d157c5e1909d3366bce098951 Mon Sep 17 00:00:00 2001
From: Martin Lang <martin.lang@mpsd.mpg.de>
Date: Wed, 19 Nov 2025 11:47:53 +0100
Subject: [PATCH 5/7] Replace external definitions with use omp_lib

Using external leads to a name mangling issue where the linker is looking for omp_get_thread_num_,
which is not defined in llvm-openmp.
---
 source/MonteCarlo/montecarlo.f90 | 1 -
 source/Tools/profiling.f90       | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/source/MonteCarlo/montecarlo.f90 b/source/MonteCarlo/montecarlo.f90
index 0851717..a189456 100644
--- a/source/MonteCarlo/montecarlo.f90
+++ b/source/MonteCarlo/montecarlo.f90
@@ -132,7 +132,6 @@ contains
       real(dblprec),dimension(natom,mensemble) :: flipprob_a ,newmmom_a,mflip
       real(dblprec),dimension(3,natom,mensemble) :: newmom_a,flipprob_g,flipprob_m
 
-      integer, external :: omp_get_thread_num
       real(dblprec) :: henergy
 
       cluster_size=0.0_dblprec
diff --git a/source/Tools/profiling.f90 b/source/Tools/profiling.f90
index b6fb189..0c5e311 100644
--- a/source/Tools/profiling.f90
+++ b/source/Tools/profiling.f90
@@ -4,6 +4,7 @@
 !> @copyright
 !> GNU Public License.
 module Profiling
+   use omp_lib
    use Parameters
 
    implicit none
@@ -157,7 +158,6 @@ contains
       logical :: parallel,init
       integer, parameter :: ncat=13 ! define timing categories
       integer :: i,ii,nprocs
-      integer, external :: omp_get_num_threads,OMP_GET_NUM_PROCS
 
       !cputime routine gives a real
       real(kind=8) :: total,total0,time,time0
